### DISCUSSION TITLE
Welcome to scribble.rs Discussions!

### DISCUSSION THREAD

**Discussion by Bios-Marcel** [scribble-rs/scribble.rs#178] [General]
## 👋 Welcome!

This is the place where you can ask for help or get answers to your questions.
This way we can prevent the issue section from being cluttered with such things and you don't have to worry about asking questions anymore. In the past, I've often seen people not wanting to create issues because they thought "it wasn't worth an issue".

So, ask questions to your hearts content!

**Comment by maquis196** [comment]
Afternoon again! I was wondering about your position on adding more languages. I see that the majority of the "latin" alphabet languages are accounted for, but do you have concerns over number of languages on the present interface for choosing them? Or is that not a problem until there is a lot more than 6 to choose from? 

I ask, because this seems like the perfect game to add Icelandic too, which obviously is niche as hell, but i think would be extremely popular considering how little choice they have (I'd also be tempted to add Welsh as well).

Or is it a case of "as soon as someone wants to make the effort of adding a language, go ahead"?

**Reply by Bios-Marcel** [reply]
Well, to be honest. I already dislike the fact that I added italian, french and dutch. Simply because I don't understand them.
What this caused, is that we now have different wordlists with vastly different amounts and maybe even quality of words that I simply can't maintain. The same goes for when I change the format or add features to the wordlist.

In general I have nothing against adding other languages, it's just that I can't see myself maintaining it.
If we had a sufficiently big userbase, this might be fine. But I don't see that right now.

I think some of the words were even automatically translated, not sure anymore.
The recent german and english wordlists have been scraped from skribbl.io though.
However, even these are not very nice imo. Especially if you don't have a difficulty grade.

**Reply by maquis196** [reply]
I recall French was automatically translated via AWS, and tbf for 5000 words id do the same for Icelandic and then get a couple of people to run the rule over them along with myself to fix obvious errors. 

In theory, if you were resigned to the fact that potentially every language known to man could/would be implemented, how would you go about that? Obviously you couldn't vouch/admin all of them, maybe its worth splitting the words from the project into seperate github accounts that can be imported at build time (so maybe per language, 1st class language support like what you have in one repo, unofficial/hobby, other). That way, you dont need to maintain them all, and perhaps people could volunteer to monitor them. 

Thats easy whilst we have a flat list, long term with more complicated word list features perhaps a different approach might be needed. With the right tests on each repo, it could be done? (I admit, I'm only a sysadmin/devops, my go knowledge is like my python, low, but happy to help with this)

**Reply by Bios-Marcel** [reply]
Well, maybe #84 is interesting in this regard. A fairly big feature I had planned, but never got around to implement.

**Reply by Bios-Marcel** [reply]
Either way, I am currently focusing a bit on UX and minor feature. On top of that device compatibility and API / stability improvements. After I feel satisfied in all those areas, I'll probably have to cleanup the HTML and CSS ^^'

So, I don't think I'd be able to get to the word galeries thing soon. But it'd be like you said with "first class" and "second class" languages.

If you want to help in other areas ofc, that means I might have more time for the big things, any contributions are generally welcome ^^

**Reply by maquis196** [reply]
Im trying to think of another project that has similar 'use-case' like this. I personally love the idea of scribble.rs being as you say, an app/frontend/experience that can run anywhere, be hosted on anything, lightweight and does its job very well. 

The words themselves being something that it consumes and is much more modular like you say in #84, if there are maintainers for a word list perhaps they can count as being 'first class' and always included in a release (like how ports get updated in freebsd for example) that way you never need to worry about maintaining anything beyond making sure at least german + english are covered yourself as "core" languages or something. 

I'm mostly thinking out loud here, language geek here so quite interested in how this would look, and haven't even touched on translations for scribble itself :). (I know some projects have a translation matrix)

**Reply by Bios-Marcel** [reply]
Maybe we should formulate this in an issue, so it doesn't go under. Cause overall I don't dislike the idea.
